PLOS/rhino
# Experimental S3 code

:warning: do not merge :warning:

Experiment code to connect rhino directly to S3, avoiding content repo.

Testing:
1. Create a new S3 bucket accessible by the plos dev role.
1. Set your secret key: `$ export AWS_SECRET_ACCESS_KEY=XXX`
1. Ensure you have an `ambra` database.
1. Start rhino to initialize the db schema.  `
$ AWS_ACCESS_KEY_ID=AKIA4SPCIUSTALRESEYQ AWS_ROLE_ARN="arn:aws:iam::429579970117:role/PLOS_DevAccess" CORPUS_BUCKET=my-test-bucket TAXONOMY_URL=https://plos.accessinn.com:9138/servlet/dh THESAURUS=plosthes.2017-2 DATABASE_URL="jdbc:mysql://localhost:3306/ambra?user=root&password=password" mvn clean package cargo:run`
1. Seed the db using the sql here: https://github.com/PLOS/molten/blob/dev/vagrant/apps/ambra/seed.sql
1. Ingest an article per https://github.com/PLOS/molten/blob/dev/vagrant/apps/ambra/README.md, specifically:
```
$ curl -X POST http://localhost:8002/v2/articles -Farchive=@pone.0174958.zip
$ curl -X POST http://localhost:8002/v2/articles/10.1371++journal.pone.0174958/revisions -Fingestion=1
```
1. You can now explore the rhini API. You can also start wombat per instructions here: https://github.com/PLOS/wombat/pull/1606


Assets are now sent back as expiring URLs:
```
$  curl -s http://localhost:8082/v2/articles/10.1371++journal.pone.0174958/ingestions/1/items.json | jq . | head
{
  "items": {
    "10.1371/journal.pone.0174958.e001": {
      "doi": "10.1371/journal.pone.0174958.e001",
      "itemType": "graphic",
      "files": {
        "original": {
          "size": 0,
          "url": "https://egh-test-rhino-direct-s3.s3.amazonaws.com/10.1371/journal.pone.0174958/1/pone.0174958.e001.tif?x-amz-security-token=FwoGZXIvYXdzEJn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaDF7%2BkUmZSRyLQTICAyLIAYzaqQWvVfAd4bFTTBE%2FEfZbw2meDl21%2BcGdT%2B0maC5QDm4nCpDCwlmGCttlh5CpK8WoPF80ZihIOmFBKr3HEnauxBR7kuCmOoS6ccdDhMpcPHvzpCr3H2F%2F9I%2FKr3ZeDXQ7F8rB03eAaLjyBvMaJdVbrpoLdxcw3C6IkZzxoc4RFcpuzXyFf5KDoEkmfqdeXs3WJg9ncX7M8Wk7%2BBPhvmxyWQyyshsjDEdP%2B5XHtJ8qOfbV6KNbTH1IMANsdauTOO%2Fw%2Bf3dzwsMKJbb7%2FMFMi1iPcobM5Giced5Pf08cu1XZC778vhHeIaOU8q2l1iK%2BPdiBiRmB9QZme%2BtzxY%3D&AWSAccessKeyId=ASIAWIBHWEZC5QTYINNC&Expires=1585183669&Signature=VMlNBzs4IQIgPTEC84WKOBA8qZE%3D"
        },
```
